BuildMacOSUniversalBinary.py

```diff
@@ -162,6 +162,32 @@ def recursive_merge_binaries(src0, src1, dst):
        the source trees
     """
 
+    # Check that all files present in the folder are of the same type and that
+    # links link to the same relative location
+    for newpath0 in glob.glob(src0+"/*"):
+        filename = os.path.basename(newpath0)
+        newpath1 = os.path.join(src1, filename)
+        if not os.path.exists(newpath1):
+            continue
+
+        if os.path.islink(newpath0) and os.path.islink(newpath1):
+            if os.path.relpath(newpath0,src0) == os.path.relpath(newpath1,src1):
```

Comment:
Shouldn't this be checking the realpath instead? Or am I misunderstanding what this does?

Reply:
realpath would be the absolute path, which wouldn't match. It should be, and is, checking the relative paths within each bundle match each other.

Reply:
It needs to check relpath because the realpath will contain the architecture name in it(as the original source for the merges are in the per architecture build folders).